explainer.md

```diff
@@ -36,193 +97,126 @@ self.addEventListener('launch', event => {
-   algorithm. (If it is fulfilled, do not continue with the navigation.)
-
-The user agent **MAY** go through the `launch` flow after going through the
-normal navigation behaviour. For example, the user agent might be displaying
-a choice asking asking if the user would like to use the `launch` flow for this
-launch and future launches. The user agent might prefer to immediately display
-the page being navigated to, instead of waiting for the user to make a choice.
-Thus the choice might be presented as a popup that the user can ignore. If
-the user requests the `launch` flow, then the event fires and the navigation
-occurs in the window that is in focus when the event completes. Note that the
-navigation might involve a POST request, in which case the POST may be handled
-twice.
-
-# Security and privacy considerations
-
-* The user agent must only fire a `launch` event for navigations to URLs inside
-  the service worker's scope, or a service worker could spy on other
-  navigations.
-* By providing the service worker with the client ID of the navigation source,
-  the target site may be able to learn a lot more information about where the
-  user came from than the usual `Referer` header.
-* Not really security or privacy, but a concern as it could be seen as taking
-  control away from the user. In particular, it could be confusing to click a
-  link and have nothing happen. Mitigations:
-  * You can already click a link and have nothing happen (as it might have an
-    onclick handler that prevents default).
-  * The user has full control if they use explicit navigation commands like
-    "Open in new tab", which do not fire the `launch` event.
+6. If [[*handlePending*]] is rejected, proceed with the normal navigation algorithm. (If it is fulfilled, do not continue with the navigation.)
+
+The user agent **MAY** go through the `launch` flow after going through the normal navigation behaviour. For example, the user agent might be displaying a choice asking asking if the user would like to use the `launch` flow for this launch and future launches. The user agent might prefer to immediately display the page being navigated to, instead of waiting for the user to make a choice.
+Thus the choice might be presented as a popup that the user can ignore. If the user requests the `launch` flow, then the event fires and the navigation occurs in the window that is in focus when the event completes. Note that the navigation might involve a POST request, in which case the POST may be handled twice.
+
+## Security and privacy considerations
+
+* The user agent must only fire a `launch` event for navigations to URLs inside the service worker's scope, or a service worker could spy on other navigations.
+* By providing the service worker with the client ID of the navigation source, the target site may be able to learn a lot more information about where the user came from than the usual `Referer` header.
```

Comment:
Do you know if they can actually read information about the site they're coming from? I have a suspicion they can't but I don't really know.

Reply:
It would be a violation of SOP if they did. I don't think historically we've ever added the ability to get a client ID for an origin you don't own, so there would be some things to think through here. Added a note.

Reply:
Cool, that's what I hoped